### DISCUSSION TITLE
Display Schematic Questions

### DISCUSSION THREAD

**Discussion by Fishbone69** [olikraus/u8g2#2293] [General]
Hi:
     I am using an SH1106 display  and the breakout board has the following schematic:
![SH1106](https://github.com/olikraus/u8g2/assets/126020608/1f74e526-9fa3-4c64-9430-f493631fcb1d)
The breakout board has a jumper for I2C or SPI.  I am planning on integrating this display on my own pcb and am trying to understand if I need all all of the circuitry that is on the breakout board.  The final design will be SPI and have a +3.3V supply. 
1.  The table in the figure shows R6, R7, and R8 are for I2C.  R6 and R7 look like pullups for D0 and D1  which I believe are the data lines for I2C.  For SPI D0 is SCLK and D1 is MOSI.  Do I still need the pullups for SPI?  I currently have them in my design and it works, but If I can remove them, that would be better.
2. What is the 662K Voltage regulator for? The supply voltage for this design is the 3.3V from an esp32.  Is this voltage regulator  necessary or a nice to have? 

Any input is appreciated!
Fish

**Comment by olikraus** [comment]
Yes, R6 and R7 are not required for SPI. 

Of course the voltage regulator is not required if you have a stable 3.3V source. But I would also consider the question, whether the ESP32 is able to provide the required current. I mean, if the ESP32 is able to provide 100mA as output and the display requires 150mA peak, then using the ESP32 3.3V output might be not good enough.
